File: binarysearch/BinarySearch.java
public class BinarySearch{

    public static  Integer findItemIndex(int[] arr,int item){
        int low = 0;
        int high = arr.length - 1;
        int mid;
        int guess;
        while(low <=high){
            mid = (low+high)/2;
            guess = arr[mid];
            if(guess==item){
                return mid;
            }else if(guess>item){
                high = mid -1;
            }else{
                low = mid + 1;
            }
        }
        return -1;
    }
    public static void main(String[] args){

        int[] arr = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20};
        int item = 66;

        Integer index = findItemIndex(arr,item);

        if(index == -1){
            System.out.println("Item not found");
        }else{
            System.out.println("Item found at index = "+index);
        }

    }

}

File: binarysearch/BinarySearch.py
#Complexity: O(log(n))
def find_item(list,item):
    low = 0
    high = len(list) - 1
    
    while low <= high: 
        mid = int((low+high)/2)
        guess = list[mid]
        if guess == item:
            return mid
        elif guess < item:
            low = mid + 1
        else:
            high = mid - 1
    return None

item  = input("Please enter the number you want to search for:")
try:
    item = int(item)
    list = [1,2,3,4,5,6,7,8,9,11,12,13,14,15,16,17]
    if None != item:
        index = find_item(list,item)
        if None != index:
            print("Found the number at index: ",index)
        else:
            print("Not found")
    else:
        print("Invalid Input !!")
except ValueError:
    print("Invalid Input")
except TypeError:
    print("Invalid Input")

